src/Uno.UI/UI/Xaml/Media/Imaging/SvgImageSource.iOSmacOS.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+using Uno.UI.Xaml.Media;
+using System.Net.Http;
+using Foundation;
+using UIKit;
+using Uno.Extensions;
+using Uno.Foundation.Logging;
+using Uno.UI.Dispatching;
+using Windows.Storage;
+using Windows.Storage.Streams;
+using CoreFoundation;
+using Windows.UI.Xaml.Shapes;
+
+namespace Windows.UI.Xaml.Media.Imaging;
+
+partial class SvgImageSource
+{
+	private protected override bool IsSourceReady => true;
+
+	private protected override bool TryOpenSourceAsync(CancellationToken ct, out Task<ImageData> asyncImage)
+	{
+		asyncImage = TryOpenSourceAsync(ct);
+		return true;
+	}
+
+	private async Task<ImageData> TryOpenSourceAsync(CancellationToken ct)
+	{
+		var imageData = await TryReadImageDataAsync(ct);
+		if (imageData.Kind == ImageDataKind.ByteArray)
+		{
+			if (_svgProvider is null)
+			{
+				this.Log().LogWarning("Uno.UI.Svg package needs to be installed to use SVG on this platform.");
+			}
+			_svgProvider?.NotifySourceOpened(imageData.ByteArray);
+		}
+		return imageData;
+	}
+
+	private async Task<ImageData> TryReadImageDataAsync(CancellationToken ct)
+	{
+		try
+		{
+			if (Stream != null)
+			{
+				Stream.Position = 0;
+				using (var data = NSData.FromStream(Stream))
+				{
+					var bytes = data.ToArray();
+					return ImageData.FromBytes(bytes);
+				}
+			}
+
+			if (FilePath.HasValue())
+			{
+				var data = NSData.FromFile(FilePath);
+				var bytes = data.ToArray();
+				return ImageData.FromBytes(bytes);
+			}
+
+			if (HasBundle)
+			{
+				return OpenSvgBundle(BundlePath);
+			}
+
+			if (Downloader == null)
+			{
+				return await DownloadSvgAsync(ct);
+			}
+			else
+			{
+				var localFileUri = await Download(ct, WebUri);
+
+				if (localFileUri == null)
+				{
+					return ImageData.Empty;
+				}
+
+				return OpenSvgBundle(localFileUri.LocalPath);
+			}
+		}
+		catch (Exception ex)
+		{
+			return ImageData.FromError(ex);
+		}
+	}
+
+	private ImageData OpenSvgBundle(string bundlePath)
+	{
+		var directoryName = global::System.IO.Path.GetDirectoryName(bundlePath);
+		var fileName = global::System.IO.Path.GetFileNameWithoutExtension(bundlePath);
+		var fileExtension = global::System.IO.Path.GetExtension(bundlePath);
+
+		var resourcePathname = NSBundle.MainBundle.GetUrlForResource(global::System.IO.Path.Combine(directoryName, fileName), fileExtension.Substring(1));
+		var data = NSData.FromUrl(resourcePathname);
+		var bytes = data.ToArray();
+		return ImageData.FromBytes(bytes);
+	}
+
+	/// <summary>
+	/// Opens the targetted image for this image use using the native iOS image downloader, using NSUrl.
+	/// </summary>
+	private async Task<ImageData> DownloadSvgAsync(CancellationToken ct)
+	{
+		if (ct.IsCancellationRequested)
+		{
+			return ImageData.Empty;
+		}
+
+		using (var url = new NSUrl(WebUri.AbsoluteUri))
+		{
+			using (var request = NSUrlRequest.FromUrl(url))
```

Comment:
If the url redirects, does iOS/macOS APIs follow redirects automatically?

Reply:
Not sure about it, but followed the same approach used for bitmap images on iOS/macOS.